src/Integration/Binding/BindingWorkflow.cs

```diff
@@ -336,7 +336,7 @@ private void UpdateDownloadedSonarQubeQualityProfile(RuleSet ruleSet, SonarQubeQ
 
         private void InitializeSolutionBindingOnUIThread(IProgressStepExecutionEvents notificationEvents)
         {
-            Debug.Assert(System.Windows.Application.Current?.Dispatcher.CheckAccess() ?? false, "Expected to run on UI thread");
```

Comment:
I've changed all of the direct usages of `System.Windows.Application.Current?.Dispatcher` to `host.UIDispatcher` for consistency.